src/foo/index.js

```diff
@@ -0,0 +1,2 @@
+export * from './foo1';
+export * from './foo2';
```

Comment:
Transpiled to:
```js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _foo = require("./foo1");

Object.keys(_foo).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _foo[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _foo[key];
    }
  });
});

var _foo2 = require("./foo2");

Object.keys(_foo2).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (key in exports && exports[key] === _foo2[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _foo2[key];
    }
  });
});
```

Reply:
Exception:
```
/app/dist/foo/index.js:25
  Object.defineProperty(exports, key, {
         ^

TypeError: Cannot redefine property: __ResetDependency__
    at Function.defineProperty (<anonymous>)
    at /app/dist/foo/index.js:25:10
    at Array.forEach (<anonymous>)
    at Object.<anonymous> (/app/dist/foo/index.js:22:20)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
```

Reply:
The exception happens because both `_foo` and `_foo2` have named exports of the same name, which are the ones added by the plugin: `__ResetDependency__`, `__RewireAPI__`, `__Rewire__`, `__set__`, `__get__`, `__GetDependency__`. Babel core handles this situation correctly because technically there is a collision between the exports, and Babel core is not aware of the special meaning of certain exports. So it appears to be a bug in babel-plugin-rewire. `export * from ...` seems to be the only syntax form affected by it since other ways of exporting are free from such a collision.